lib/gitlib.sh

```diff
@@ -201,3 +201,24 @@ gitlib::push_master () {
   logecho -n "Pushing$dryrun_flag master branch: "
   logrun -s git push$dryrun_flag origin master || return 1
 }
+
+##############################################################################
+# Ensure running with a synced repo.
+gitlib::repo_state () {
+  # Find remote
+  local remote=$(git -C $TOOL_ROOT remote -v |\
+                 awk '/kubernetes\/release(.git)* \(fetch\)/ {print $1}')
+  local commit=$(git -C $TOOL_ROOT \
+                     ls-remote --heads $remote refs/heads/master | cut -f1)
+
+  logecho -n "Checking $TOOL_ROOT state: "
+  if git -C $TOOL_ROOT branch --contains $commit >/dev/null 2>&1; then
```

Comment:
this will return 0 even if no branches are listed. need to check if stdout is nonempty.